src/component/variantPage/BasicInfo.tsx

```diff
@@ -1,127 +1,439 @@
+    }
+
+    private parseHgvscFromTranscriptConsequenceSummary(
+        transcript: TranscriptConsequenceSummary
+    ) {
+        if (transcript.hgvsc) {
+            let hgvsc = transcript.hgvsc;
+            let startIndex = hgvsc.indexOf('c.');
+            return startIndex !== -1 ? hgvsc.substr(startIndex) : null;
+        }
+        return null;
+    }
+
+    // find the target variant by alteration
+    private findOncokbVariant(
+        oncokb: Alteration[] | undefined
+    ): OncogeneTsg | null {
+        if (oncokb) {
+            const alterationName = this.getAlteration(this.props.annotation);
+            let oncogeneTsg: OncogeneTsg = {};
+            // find the variant with same alteration
+            const targetVariant = _.find(oncokb, variant => {
+                if (alterationName && variant.alteration === alterationName) {
+                    if (
+                        variant.gene &&
+                        (variant.gene.oncogene || variant.gene.tsg)
+                    ) {
+                        return true;
+                    }
+                }
+                return false;
+            });
+            if (targetVariant) {
+                oncogeneTsg = {
+                    oncogene: targetVariant.gene.oncogene,
+                    tsg: targetVariant.gene.tsg,
+                };
+            }
+            return oncogeneTsg;
+        }
+        return null;
+    }
+
+    private getOncogene(oncogeneData: OncogeneTsg | null) {
+        if (oncogeneData) {
+            return oncogeneData.oncogene === true ? 'Oncogene' : null;
+        } else {
+            return null;
+        }
+    }
+
+    private getTsg(oncogeneData: OncogeneTsg | null) {
+        if (oncogeneData) {
+            return oncogeneData.tsg === true ? 'TSG' : null;
+        } else {
+            return null;
+        }
+    }
+
+    private getAlteration(annotation: VariantAnnotationSummary | undefined) {
+        if (annotation && annotation.transcriptConsequenceSummary) {
+            let alteration = annotation.transcriptConsequenceSummary.hgvspShort;
+            let startIndex = alteration.indexOf('p.') + 2; // + 2 will exclude the 'p.' in the result
+            return startIndex !== 1 ? alteration.substr(startIndex) : null; // !== -1 means there is no 'p.' in the string, here because the startIndex was +2 before, so have !== 1 instead
+        } else {
+            return null;
+        }
+    }
+
+    private haveTranscriptTable(
+        annotation: VariantAnnotationSummary | undefined
+    ): boolean {
+        if (
+            annotation &&
+            annotation.transcriptConsequenceSummary !== undefined &&
+            annotation.transcriptConsequenceSummaries !== undefined &&
+            annotation.transcriptConsequenceSummaries.length > 1
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    private transcriptsButton(isOpened: boolean) {
+        if (isOpened === false) {
```

Comment:
you should have this condition for your `<span>` element, so that you don't need to duplicate the entire `<Button>` element.